### ISSUE TITLE
Make gradle-home-cache-cleanup: true the default

### ISSUE THREAD

**Issue by martinbonnin** [gradle/gradle-build-action#630]
Hi 👋 Thanks for saving our build times ❤️ Would it make sense to include `gradle-home-cache-cleanup: true` by default and have the option opt-out? Sounds like this would be a reasonable default to expect?

**Comment by bigdaz** [comment]
@martinbonnin Yes this is the plan longer term, but the cache-cleanup was released as "experimental" and I didn't feel ready to enable it by default. Enabling by default will be in an upcoming release.

**Comment by bigdaz** [comment]
There are a few issues we'll need to deal with before enabling this by default:
- #426 
- #603 
- #858
